runtime-modules/membership/src/lib.rs

```diff
@@ -182,6 +186,35 @@ pub struct BuyMembershipParameters<AccountId, MemberId> {
     pub referrer_id: Option<MemberId>,
 }
 
+#[derive(Encode, Decode, Clone, PartialEq, Debug, Eq, TypeInfo)]
+pub struct GiftMembershipParameters<AccountId, Balance> {
+    /// New member root account.
+    pub root_account: AccountId,
+
+    /// New member controller account.
+    pub controller_account: AccountId,
+
+    /// New member handle.
+    pub handle: Option<Vec<u8>>,
```

Comment:
This could potentially open up doors for unbounded input type attacks, shouldn't we use [BoundedVec](https://paritytech.github.io/substrate/master/frame_support/struct.BoundedVec.html)

Reply:
That is a very good point. The `BuyMembershipParameters` and `InviteMembershipParameters` have the same problem. As a matter of fact this may be the case for other calls in more than one pallet.

Now I see the benchmarks use an upper bound `const MAX_BYTES: u32 = 16384;` for inputs like `Vec<u8>`. I think if that would fill a block and the weight function of the call uses that as a parameter, then the attacker will be paying a large fees for the attack. Is that sufficient to mitigate the attack.

Although the handle and metadata are not stored in state, It probably still make sense to test query-node if it can handle large values.

Reply:
This issue about `Vec<u8>` is present all over the code base, maybe it is worth to open an issue in order to find a definite solution (provided that it is not already present)
I will take care of the issue. Let's leave this part like this for the moment